mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala

```diff
@@ -287,6 +290,27 @@ class LinearSVCModel private[classification] (
   @Since("2.2.0")
   def setWeightCol(value: Double): this.type = set(threshold, value)
 
+  private var trainingSummary: Option[LinearSVCTrainingSummary] = None
+
+  private[classification]
+  def setSummary(summary: Option[LinearSVCTrainingSummary]): this.type = {
+    this.trainingSummary = summary
+    this
+  }
+
+  /**
+   * Gets summary of model on training set. An exception is
+   * thrown if `trainingSummary == None`.
+   */
+  @Since("2.2.0")
+  def summary: LinearSVCTrainingSummary = trainingSummary.getOrElse(
+    throw new SparkException("No training summary available for this LinearSVCModel")
```

Comment:
I know other places has the same code here. But I don't think throwing an exception here is proper. If summary is not defined, we should simply return none, or keep this field as Option[LinearSVCTrainingSummary] to make that expected by the users. Need to confirm with committers.

Reply:
These should be marked Since 3.0.0 too.